### PR TITLE
Feature/ia tickets 27 72 user preferences

### LINKED ISSUE #27: As an authenticated user, I want to be able to save my location
**__User Story Description__**
As a user
I want to be able to save my location (country & state or province) to my profile 
So I can use this information to fill out forms quicker

**__Steps to Follow (optional)__**
- [ ] TBD
- [ ] Additional steps as necessary

**__Additional Considerations__**
Any supplemental information including unresolved questions, links to external resources, screenshots, etc.

### LINKED ISSUE #72: As a registered user I want to set preferences for certain fields so I can fill out each carbon emission calculator faster
**__User Story Description__**
As a registered user,
I want to set preferences for certain fields
So I can fill out each carbon emission calculator faster

**__Steps to Follow (optional)__**
- [ ] `account/preferences` should route to Preferences page
- [ ] Preferences page should include fields for Country, State (Province), and Vehicle Info
- [ ] `Save Preferences` button should be disabled until an input field is filled out
- [ ] Input field should contain a valid value for that type
- [ ] Additional steps as necessary

**__Additional Considerations__**
Rough wireframe for reference (feel free to adjust anything this is just a rough template of what to do)
**Add a `Save Preferences` button (forgot to add that to the wireframe)**
PreferencesPage

### PR DESCRIPTION
**Pull request by ialej001**
Please advise:

**You need to run `npm install` in order for the prisma schema on your machine to update properly. If you do not do this, you'll get some errors related to unknown fields.**

A big one for y'all!

Features:
- Added user ability to configure their preferences
  - Location info (`country` is mandatory, `state` is optional)
  - Unit preference between metric and imperial
  - Feedback given through the change of the button:
    - Green button with checkmark on successful save
    - Blue button all other times. Will revert back to blue on any field change.
- Vehicle controls:
  - Can save cars using a similar style form to the travel/driving calculator from within a modal
  - Can remove a saved car via a pop-up modal
    - If a car to be removed is the primary selected (and saved) car, it will clear that `id` field in the DB
  - Able to select the primary car once a car has been saved
  - Renders a message when there are no saved cars

There are also some non related fixes in here that I did not want to split out:
- Account level layout
  - This includes a `ControlPanel` component that has three buttons that route to different `/account` pages
  - This gives the `/account` route and all subsequent routes the same layout, offering consistency
- Assorted images that were broken by file name changes in other PRs are now fixed

Closes #27 #72

### CODE REVIEW

File: src/pages/account/preferences.tsx [hunk 1]
```diff
@@ -1,20 +1,180 @@
...
+  const [showModal, setShowModal] = useState(false);
+
+  const handleToggleModal = () => {
+    setShowModal((old) => !old);
+  };
+
+  const handleLocationUpdate = () => {
+    mutateLocation({ country, state });
+  };
+
```
Comment (TiciaD) [inline src/pages/account/preferences.tsx:67]: Can we add some kind of confirmation message to the user on successful update? Maybe like a toast or something?

File: src/components/Account/Preferences/UserVehicles.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    }
+    return;
+  };
+
+  if (!userVehicleData)
+    return (
+      <div className="flex justify-center items-center">
+        <Spinner />
+      </div>
+    );
```
Comment (TiciaD) [inline src/components/Account/Preferences/UserVehicles.tsx:81]: All I'm getting is this Spinner, when I first load the preferences page and even after I try to add a Vehicle.

Screenshot 2022-09-06 204501

**Review by TiciaD — COMMENTED** [review]
On top of the Spinner issue, I'm getting an error when I try to add a Vehicle:
```
Something went wrong Error: 
[0] Invalid `prisma.user.findFirst()` invocation:
[0] 
[0] {
[0]   select: {
[0]     primaryVehicleId: true,
[0]     ~~~~~~~~~~~~~~~~
[0] ?   id?: true,
[0] ?   name?: true,
[0] ?   email?: true,
[0] ?   emailVerified?: true,
[0] ?   image?: true,
[0] ?   country?: true,
[0] ?   state?: true,
[0] ?   unitPref?: true,
[0] ?   accounts?: true,
[0] ?   sessions?: true,
[0] ?   vehicles?: true,
[0] ?   electricityUse?: true,
[0] ?   fuelUsed?: true,
[0] ?   flights?: true,
[0] ?   _count?: true
[0]   },
[0]   where: {
[0]     id: 'cl7qvrlab0006vem3xs1lqowx'
[0]   }
[0] }
[0] 
[0] 
[0] Unknown field `primaryVehicleId` for select statement on model User. Available options are listed in green.
```

**Comment by TiciaD** [comment]
> On top of the Spinner issue, I'm getting an error when I try to add a Vehicle:
> 
> ```
> Something went wrong Error: 
> [0] Invalid `prisma.user.findFirst()` invocation:
> [0] 
> [0] {
> [0]   select: {
> [0]     primaryVehicleId: true,
> [0]     ~~~~~~~~~~~~~~~~
> [0] ?   id?: true,
> [0] ?   name?: true,
> [0] ?   email?: true,
> [0] ?   emailVerified?: true,
> [0] ?   image?: true,
> [0] ?   country?: true,
> [0] ?   state?: true,
> [0] ?   unitPref?: true,
> [0] ?   accounts?: true,
> [0] ?   sessions?: true,
> [0] ?   vehicles?: true,
> [0] ?   electricityUse?: true,
> [0] ?   fuelUsed?: true,
> [0] ?   flights?: true,
> [0] ?   _count?: true
> [0]   },
> [0]   where: {
> [0]     id: 'cl7qvrlab0006vem3xs1lqowx'
> [0]   }
> [0] }
> [0] 
> [0] 
> [0] Unknown field `primaryVehicleId` for select statement on model User. Available options are listed in green.
> ```

SOLVED: Ivan pair programmed with me, since some changes were made to the schema, I had to run `npm install` again to get it to update the database properly and it worked fine. Leaving this comment here in case anyone runs into the same error I did!

File: src/pages/account/preferences.tsx (hunk 1 above)
Comment (ialej001) [reply src/pages/account/preferences.tsx:67]: Updated that. Buttons now change color and text (with icon) on successful save

**Review by TiciaD — APPROVED** [review]
Thanks for adding that Toast for user feedback, I think it helps the UX a little more, approved!

**Review by amandaalexandre — APPROVED** [review]

**Pull request merged**